src/app/src/test/kotlin/org/imperial/mrc/hint/unit/clients/FuelFlowClientTests.kt

```diff
@@ -0,0 +1,98 @@
+package org.imperial.mrc.hint.unit.clients
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.nhaarman.mockito_kotlin.doReturn
+import com.nhaarman.mockito_kotlin.mock
+import org.assertj.core.api.Assertions
+import org.imperial.mrc.hint.AppProperties
+import org.imperial.mrc.hint.clients.FuelFlowClient
+import org.imperial.mrc.hint.models.ErrorReport
+import org.imperial.mrc.hint.models.Errors
+import org.junit.jupiter.api.Test
+import org.springframework.http.ResponseEntity
+
+class FuelFlowClientTests
+{
+    private val data = ErrorReport(
+            "test.user@example.com",
+            "Kenya",
+            "Kenya2022",
+            "Model",
+            "123",
+            "1234",
+            mapOf("spectrum" to "spectrum123", "summary" to "summary123", "coarse_output" to "coarse123" ),
+            listOf(
+                    Errors("#65ae0d095ea", "test error msg", "fomot-hasah-livad"),
+                    Errors("#25ae0d095e1", "test error msg2", "fomot-hasah-livid")
+            ),
+            "",
+            "test steps",
+            "test agent",
+            mapOf(
+                    "naomi" to "v1",
+                    "hintr" to "v2",
+                    "rrq" to "v3",
+                    "traduire" to "v4",
+                    "hint" to "v5"
+            ),
+            "2021-10-12T14:07:22.759Z"
+    )
+
+    @Test
+    fun `can serialize request body`()
+    {
+        class LocalFuelFlowClient(objectMapper: ObjectMapper,
+                                  appProperties: AppProperties
+        ): FuelFlowClient(objectMapper, appProperties)
```

Comment:
This looks suspicious but I have no comment.

Reply:
It is a bit smelly, as I say, but it's quite a tricky thing to mock.

Reply:
With your explanation, it could be something like below.

```
    @AfterEach
    fun tearDown() {
        FuelManager.instance.reset()
    }
```

Reply:
https://github.com/mrc-ide/hint/pull/691

Reply:
I did consider doing that , but to be honest, I still prefer this approach. Firstly because I think the controller test is wrong place to test the serialization of the ErrorReport class (it's not the controller that's doing that).  Secondly, because setting the FuelManager.instance makes even more assumptions about the internal guts of the Fuel clients than my subclass. Finally out of paranoia -  `tearDown` ought to be able to clear the mocked instance, but I think it's probably better not to mock it in the first place.